### PR TITLE
create zh-trans content/zh/docs/reference/access-authn-authz/rbac.md

### PR DESCRIPTION
**Pull request by liyinda246**
Add rbac.md file, I found that there is no corresponding Chinese translation in the official website, so I translated rbac.md in Chinese, I hope it will be helpful, thanks

### CODE REVIEW

**Comment by liyinda246** [comment]
@tengqm I have recreated PR

File: content/zh/docs/reference/access-authn-authz/rbac.md [hunk 1]
```diff
@@ -0,0 +1,2137 @@
+translator:
+- liyinda246
+reviewers:
+- tengqm
+title: 使用 RBAC 鉴权
+content_template: templates/concept
+weight: 70
+---
```
Comment (tengqm) [inline content/zh/docs/reference/access-authn-authz/rbac.md]: Please replace line 1-4 with "---". We don't need to sign the translator/reviewer names. The page preview doesn't look good at the moment.

Comment (liyinda246) [reply content/zh/docs/reference/access-authn-authz/rbac.md]: ok

Comment (liyinda246) [reply content/zh/docs/reference/access-authn-authz/rbac.md]: How to complete the approval? can you approval it?

Comment (tengqm) [reply content/zh/docs/reference/access-authn-authz/rbac.md]: please delete line 1-3, the preview is still incorrect: https://deploy-preview-19276--kubernetes-io-master-staging.netlify.com/zh/docs/reference/access-authn-authz/rbac/

Comment (liyinda246) [reply content/zh/docs/reference/access-authn-authz/rbac.md]: I have deleted line 1-3

File: content/zh/docs/reference/access-authn-authz/rbac.md [hunk 2]
```diff
@@ -0,0 +1,2136 @@
...
+改变现有绑定对象中 `roleRef` 字段的内容，必须删除并
+重新创建绑定对象。这种限制有两个主要原因：
+
+1.关于不同角色（Role）的绑定是完全不一样的。更改 `roleRef`
+ 需要删除/重建绑定，确保要赋予绑定的完整主体列表是新
+的角色（而不是只是启用修改 `roleRef` 在不验证所有现有
+主体的情况下的，应该授予新角色对应的权限）。
+2.使得 `roleRef` 不可以改变现有绑定主体用户的 `update` 权限，
+这样可以让它们能够管理主体列表，而不能更改授予这些主体相关
+的角色。
```
Comment (tengqm) [inline content/zh/docs/reference/access-authn-authz/rbac.md]: 因为缩进问题，这个列表显示的不正确。

File: content/zh/docs/reference/access-authn-authz/rbac.md [hunk 3]
```diff
@@ -0,0 +1,2136 @@
...
+
+命令 `kubectl auth reconcile` 可以创建或者更新包含 RBAC 对象的清单文件，
+处理删除和重新创建绑定对象如果需要更改它们引用的角色。
```
Comment (tengqm) [inline content/zh/docs/reference/access-authn-authz/rbac.md]: 并且在必要的情况下删除和重新创建绑定对象，以改变所引用的角色。

File: content/zh/docs/reference/access-authn-authz/rbac.md [hunk 4]
```diff
@@ -0,0 +1,2136 @@
...
+更多相关信息请参照[command usage and examples](#kubectl-auth-reconcile)
```
Comment (tengqm) [inline content/zh/docs/reference/access-authn-authz/rbac.md]:
```suggestion
更多相关信息请参照[命令用法和示例](#kubectl-auth-reconcile)
```

File: content/zh/docs/reference/access-authn-authz/rbac.md [hunk 5]
```diff
@@ -0,0 +1,2136 @@
...
+  name: pod-and-pod-logs-reader
+rules:
+- apiGroups: [""]
+  resources: ["pods", "pods/log"]
+  verbs: ["get", "list"]
+```
+-->
+### 对资源的引用
+
+大多数资源都是使用名称的字符串表示，比如 "pods" 出现在相关的 API endpoint 的 URL 之中，
```
Comment (tengqm) [inline content/zh/docs/reference/access-authn-authz/rbac.md]:
```suggestion
大多数资源都是使用名称的字符串表示，例如在相关的 API 末端的 URL 之中出现的  "pods" ，
```

File: content/zh/docs/reference/access-authn-authz/rbac.md [hunk 6]
```diff
@@ -0,0 +1,2136 @@
...
+然而有一些 Kubernetes API 频及 "subresource"子资源，例如 pod 的日志，鉴权它相关的 endpoint URL 如下：
```
Comment (tengqm) [inline content/zh/docs/reference/access-authn-authz/rbac.md]:
```suggestion
然而有一些 Kubernetes API 涉及 "子资源（subresources）"，例如 pod 的日志。Pod 日志相关的端点 URL 如下：
```

File: content/zh/docs/reference/access-authn-authz/rbac.md [hunk 7]
```diff
@@ -0,0 +1,2136 @@
...
+
+```http
+GET /api/v1/namespaces/{namespace}/pods/{name}/log
+```
+
+在这种情况下，"pods" 是命名空间的资源，而 "log" 是 pods 的子资源。在 RBAC 角色中，
```
Comment (tengqm) [inline content/zh/docs/reference/access-authn-authz/rbac.md]:
```suggestion
在这种情况下，"pods" 是有命名空间的资源，而 "log" 是 pods 的子资源。在 RBAC 角色中，
```

File: content/zh/docs/reference/access-authn-authz/rbac.md [hunk 8]
```diff
@@ -0,0 +1,2136 @@
...
+  namespace: default
+  name: configmap-updater
+rules:
+- apiGroups: [""]
+  resources: ["configmaps"]
+  resourceNames: ["my-configmap"]
+  verbs: ["update", "get"]
+```
+
+需要注意的是，`create` 请求不能被 resourceName 限制，因为不知道对象名称的
```
Comment (tengqm) [inline content/zh/docs/reference/access-authn-authz/rbac.md]:
```suggestion
需要注意的是，`create` 请求不能被 resourceName 限制，因为在鉴权时还不知道对象名称。
```

File: content/zh/docs/reference/access-authn-authz/rbac.md [hunk 9]
```diff
@@ -0,0 +1,2136 @@
...
+# These rules will be added to the "monitoring" role.
+rules:
+- apiGroups: [""]
+  resources: ["services", "endpoints", "pods"]
+  verbs: ["get", "list", "watch"]
+```
+-->
+### Aggregated ClusterRoles
+
+从 1.9 开始，集群角色（ClusterRole）可以通过使用 `aggregationRule` 的方式并组合其他 ClusterRoles 来创建。
```
Comment (tengqm) [inline content/zh/docs/reference/access-authn-authz/rbac.md]: 全篇文章中，第一次提到“集群角色”时，可以添加“（ClusterRole）”，后续再次提及时，不要再加括号！请检查角色，角色绑定，集群角色绑定等。

File: content/zh/docs/reference/access-authn-authz/rbac.md [hunk 10]
```diff
@@ -0,0 +1,2136 @@
...
+聚合集群角色（ClusterRole）的权限是由控制器管理，并通过标记规则和提供标签选择器来匹配 ClusterRoles。
```
Comment (tengqm) [inline content/zh/docs/reference/access-authn-authz/rbac.md]:
```suggestion
聚合集群角色（ClusterRole）的权限是由控制器管理的，方法是通过过滤与标签选择器匹配的 ClusterRules，并将其中的权限进行组合。
```

File: content/zh/docs/reference/access-authn-authz/rbac.md [hunk 11]
```diff
@@ -0,0 +1,2136 @@
...
+metadata:
+  name: monitoring
+aggregationRule:
+  clusterRoleSelectors:
+  - matchLabels:
+      rbac.example.com/aggregate-to-monitoring: "true"
+rules: [] # 具体规则由控制器管理器自动填写。
+```
+
+创建一个标签选择器来匹配的 ClusterRole 上的规则，它将成为聚合集群角色（ClusterRole）。在这种情况下，
```
Comment (tengqm) [inline content/zh/docs/reference/access-authn-authz/rbac.md]:
```suggestion
创建一个与标签选择器匹配的 ClusterRole 之后，其上定义的规则将成为聚合集群角色（ClusterRole）的一部分。在这种情况下，
```

File: content/zh/docs/reference/access-authn-authz/rbac.md [hunk 12]
```diff
@@ -0,0 +1,2136 @@
...
+可以通过创建另一个具有相同标签为 `rbac.example.com/aggregate-to-monitoring: true` 的
```
Comment (tengqm) [inline content/zh/docs/reference/access-authn-authz/rbac.md]:
```suggestion
通过创建一个新的、标签同样为 `rbac.example.com/aggregate-to-monitoring: true` 的
```

File: content/zh/docs/reference/access-authn-authz/rbac.md [hunk 13]
```diff
@@ -0,0 +1,2136 @@
...
+ ClusterRole 作为 "monitoring" 集群角色（ClusterRole）。
```
Comment (tengqm) [inline content/zh/docs/reference/access-authn-authz/rbac.md]:
```suggestion
 ClusterRole，新的规则可被添加到 "monitoring" 集群角色中。
```

File: content/zh/docs/reference/access-authn-authz/rbac.md [hunk 14]
```diff
@@ -0,0 +1,2136 @@
...
+  verbs: ["get", "list", "watch"]
+- apiGroups: ["batch", "extensions"]
+  resources: ["jobs"]
+  verbs: ["get", "list", "watch", "create", "update", "patch", "delete"]
+```
+-->
+#### 角色（Role）示例
+
+在以下示例中，我们仅截取展示了 `rules` 对应部分，
+允许读取 "pods" 资源的在 {{< glossary_tooltip text="API Group" term_id="api-group" >}}下的权限:
```
Comment (tengqm) [inline content/zh/docs/reference/access-authn-authz/rbac.md]:
```suggestion
允许读取在核心 {{< glossary_tooltip text="API 组" term_id="api-group" >}}下的 Pods:
```

File: content/zh/docs/reference/access-authn-authz/rbac.md [hunk 15]
```diff
@@ -0,0 +1,2136 @@
...
+
+```yaml
+rules:
+- apiGroups: [""]
+  resources: ["pods"]
+  verbs: ["get", "list", "watch"]
+```
+
+允许 读/写 在 "extensions" 和 "apps" API groups 中 "deployments" 的相关资源:
```
Comment (tengqm) [inline content/zh/docs/reference/access-authn-authz/rbac.md]:
```suggestion
允许读/写在 "extensions" 和 "apps" API 组中的 "deployments" 资源：
```

File: content/zh/docs/reference/access-authn-authz/rbac.md [hunk 16]
```diff
@@ -0,0 +1,2136 @@
...
+
+```yaml
+rules:
+- apiGroups: ["extensions", "apps"]
+  resources: ["deployments"]
+  verbs: ["get", "list", "watch", "create", "update", "patch", "delete"]
+```
+
+允许读取 "pods" 和 读/写 "jobs" :
```
Comment (tengqm) [inline content/zh/docs/reference/access-authn-authz/rbac.md]:
```suggestion
允许读取 "pods" 和读/写 "jobs" :
```

File: content/zh/docs/reference/access-authn-authz/rbac.md [hunk 17]
```diff
@@ -0,0 +1,2136 @@
...
+
+Allow "GET" and "POST" requests to the non-resource endpoint "/healthz" and all subpaths (must be in a `ClusterRole` bound with a `ClusterRoleBinding` to be effective):
+
+```yaml
+rules:
+- nonResourceURLs: ["/healthz", "/healthz/*"] # '*' in a nonResourceURL is a suffix glob match
+  verbs: ["get", "post"]
+```
+-->
+允许读取名称为 "my-config"的 `ConfigMap` （需要通过 `RoleBinding` 绑定去限制单一 `ConfigMap` 在 namespace 上）:
```
Comment (tengqm) [inline content/zh/docs/reference/access-authn-authz/rbac.md]:
```suggestion
允许读取名称为 "my-config"的 `ConfigMap` （需要通过 `RoleBinding` 绑定带某名字空间中特定的 `ConfigMap`）：
```

File: content/zh/docs/reference/access-authn-authz/rbac.md [hunk 18]
```diff
@@ -0,0 +1,2136 @@
...
+
+```yaml
+rules:
+- apiGroups: [""]
+  resources: ["configmaps"]
+  resourceNames: ["my-config"]
+  verbs: ["get"]
+```
+
+允许读取在 core group 中 "nodes" 资源（因为 `Node` 是集群范围的，所以需要 `ClusterRole` 绑定 `ClusterRoleBinding` 才可以有效）
```
Comment (tengqm) [inline content/zh/docs/reference/access-authn-authz/rbac.md]:
```suggestion
允许读取在核心组中的 "nodes" 资源（因为 `Node` 是集群范围的，所以需要 `ClusterRole` 绑定到 `ClusterRoleBinding` 才生效）
```

File: content/zh/docs/reference/access-authn-authz/rbac.md [hunk 19]
```diff
@@ -0,0 +1,2136 @@
...
+
+```yaml
+rules:
+- apiGroups: [""]
+  resources: ["nodes"]
+  verbs: ["get", "list", "watch"]
+```
+
+允许 "GET" 和 "POST" 请求在非资源端点 "/healthz" 和其子路径（必须在 `ClusterRole` 绑定 `ClusterRoleBinding` 才有效）
```
Comment (tengqm) [inline content/zh/docs/reference/access-authn-authz/rbac.md]:
```suggestion
允许在非资源端点 "/healthz" 和其子路径上发起 "GET" 和 "POST" 请求（必须在 `ClusterRole` 绑定 `ClusterRoleBinding` 才生效）
```

File: content/zh/docs/reference/access-authn-authz/rbac.md [hunk 20]
```diff
@@ -0,0 +1,2136 @@
...
+[Service Accounts](/docs/tasks/configure-pod-container/configure-service-account/) have usernames with the `system:serviceaccount:` prefix and belong
+to groups with the `system:serviceaccounts:` prefix.
+-->
+### 对主体的引用
+
+`RoleBinding` 或者 `ClusterRoleBinding` 需要绑定角色（Role）到 *主体*。
+主体可以是组，用户或者服务账户。
+
+用户是由字符串表示，它们可以是普通的用户名，像 "alice"，或者是
+邮件格式 "bob@example.com"，或者是数字ID。由 Kubernetes 管理员配置[authentication modules](/docs/reference/access-authn-authz/authentication/)
```
Comment (tengqm) [inline content/zh/docs/reference/access-authn-authz/rbac.md]:
```suggestion
邮件格式 "bob@example.com"，或者是数字ID。由 Kubernetes 管理员配置[身份认证模块](/docs/reference/access-authn-authz/authentication/)
```

File: content/zh/docs/reference/access-authn-authz/rbac.md [hunk 21]
```diff
@@ -0,0 +1,2136 @@
...
+需要的格式。在 RBAC 鉴权系统中可以不需要任何特定的格式。但是前缀 `system:` 是 Kubernetes 系统保留的，
```
Comment (tengqm) [inline content/zh/docs/reference/access-authn-authz/rbac.md]:
```suggestion
需要的格式。RBAC 鉴权系统不对格式作任何要求，但是前缀 `system:` 是 Kubernetes 系统保留的，
```

File: content/zh/docs/reference/access-authn-authz/rbac.md [hunk 22]
```diff
@@ -0,0 +1,2136 @@
...
+所以管理员要确保配置的用户名不能出现上述前缀格式。
+
+组是 Kubernetes 现在提供的一种身份验证模块，与用户一样，组的字符串没有格式要求，
```
Comment (tengqm) [inline content/zh/docs/reference/access-authn-authz/rbac.md]:
```suggestion
用户组信息是 Kubernetes 现在提供的一种身份验证模块，与用户一样，对组的字符串没有格式要求，
```

File: content/zh/docs/reference/access-authn-authz/rbac.md [hunk 23]
```diff
@@ -0,0 +1,2136 @@
...
+只是不能使用保留的前缀 `system:` 。
+
+[Service Accounts](/docs/tasks/configure-pod-container/configure-service-account/) 是有前缀 `system:serviceaccount:` 的用户名和
```
Comment (tengqm) [inline content/zh/docs/reference/access-authn-authz/rbac.md]:
```suggestion
[服务账号](/docs/tasks/configure-pod-container/configure-service-account/) 的用户名前缀为`system:serviceaccount:`，
```

File: content/zh/docs/reference/access-authn-authz/rbac.md [hunk 24]
```diff
@@ -0,0 +1,2136 @@
...
+所属指向 `system:serviceaccounts:` 的组。
```
Comment (tengqm) [inline content/zh/docs/reference/access-authn-authz/rbac.md]:
```suggestion
属于前缀为 `system:serviceaccounts:` 的用户组。
```

File: content/zh/docs/reference/access-authn-authz/rbac.md (hunk 9 above)
Comment (liyinda246) [reply content/zh/docs/reference/access-authn-authz/rbac.md]: 好的 已经替换

File: content/zh/docs/reference/access-authn-authz/rbac.md [hunk 25]
```diff
@@ -0,0 +1,2136 @@
...
+  name: configmap-updater
+rules:
+- apiGroups: [""]
+  resources: ["configmaps"]
+  resourceNames: ["my-configmap"]
+  verbs: ["update", "get"]
+```
+
+需要注意的是，`create` 请求不能被 resourceName 限制，因为在鉴权时还不知道对象名称。
+鉴权时间。另一个例外是 `deletecollection`。
```
Comment (tengqm) [inline content/zh/docs/reference/access-authn-authz/rbac.md]:
```suggestion
另一个例外是 `deletecollection`。
```

File: content/zh/docs/reference/access-authn-authz/rbac.md [hunk 26]
```diff
@@ -0,0 +1,2136 @@
...
+    # Add these permissions to the "view" default role.
+    rbac.authorization.k8s.io/aggregate-to-view: "true"
+rules:
+- apiGroups: ["stable.example.com"]
+  resources: ["crontabs"]
+  verbs: ["get", "list", "watch"]
+```
+-->
+
+默认的面向用户的角色（如下所述）使用 ClusterRole 聚合。这让管理者可以使用规则属性自定义资源，
```
Comment (tengqm) [inline content/zh/docs/reference/access-authn-authz/rbac.md]:
```suggestion
默认的面向用户的角色（如下所述）使用 ClusterRole 聚合。这使得管理者可以为自定义资源设置使用规则属性，
```

File: content/zh/docs/reference/access-authn-authz/rbac.md [hunk 27]
```diff
@@ -0,0 +1,2136 @@
...
+比如在默认角色上，由 CustomResourceDefinitions 或 Aggregated API servers 提供的服务。
```
Comment (tengqm) [inline content/zh/docs/reference/access-authn-authz/rbac.md]:
```suggestion
比如通过 CustomResourceDefinitions 或聚合 API 服务器为默认角色提供的服务。
```

**Comment by tengqm** [comment]
Preview 仍然有问题，请自行检查并修改。
https://deploy-preview-19276--kubernetes-io-master-staging.netlify.com/zh/docs/reference/access-authn-authz/rbac/

**Comment by liyinda246** [comment]
> Preview 仍然有问题，请自行检查并修改。
> https://deploy-preview-19276--kubernetes-io-master-staging.netlify.com/zh/docs/reference/access-authn-authz/rbac/

Permissive RBAC Permissions 格式问题已经更改

**Comment by tengqm** [comment]
文内有些表格的第二列需要部分翻译。

**Comment by liyinda246** [comment]
@tengqm 表中的第二列已经翻译

**Comment by liyinda246** [comment]
@tengqm  您好  请问如何才可以merge上？我这还有哪些没有符合规则的地方吗？打扰了

**Comment by tengqm** [comment]
@liyinda246 this looks good to me now, waiting for reviews from some other approvers.

**Comment by liyinda246** [comment]
@tengqm 现在reviewers列表只有您，我需要把哪位添加到reviewers列表中吗？

**Comment by tengqm** [comment]
please be patient while waiting your PR to be reviewed/approved.

**Pull request merged**